lib/rouge/lexer.rb

```diff
@@ -126,7 +137,16 @@ def demo(arg=:absent)
 
       # @return a list of all lexers.
       def all
-        @all ||= registry.values.uniq
+        return @all if defined? @all
+        vals = registry.values.uniq
```

Comment:
Since `vals` is just cosmetic, this can be iterated through once:
```ruby
@all = []
registry.each_value do |val|
  next if val.nil?
  next val unless val.is_a?(Lexers::ProxyLexer)

  # Object.const_get("Rouge::Lexers::" + v.name.to_s)
  @all << Lexers.const_missing v.name
end
```

Reply:
Yeah, this needs tidying up. In terms of your solution:

1. we want to memoise `@all` to avoid doing this loop more than necessary;
2. `registry` contains multiple values that are the same and so we need to clean out the duplicates;
3. this isn't mapping over `registry` so the result of `next val` will be to throw that value away.

Reply:
The more transparent struct makes this code no longer necessary.